### ISSUE TITLE
Recommended way to conform `JSON` to `Hashable`

### ISSUE THREAD

**Issue by BenShutt** [SwiftyJSON/SwiftyJSON#1137]
### What did you do?

How would you recommend conforming `JSON` to `Hashable?
My use case looks something like this:

```swift
struct MyModel: Equatable, Hashable, Codable {
    var name: String
    var value: Int
    var json: JSON // Not Hashable error
}
```

### What did you expect to happen?

If `JSON` already conformed to `Hashable` like it does `Codable` that would be most desirable.
I could extend `JSON` to conform to `Hashable` in my project, but am interested to hear what you recommend.

### Environment

 - SwiftyJSON: *5.0.1*
 - Xcode Version: *14.3.1*
 - Swift Version: *5.8.1*
 - Swift Package Manager - *Swift 5.8.1*

**Comment by Wejua** [comment]
YES, This will be a very nice feature that i can use it directly in ForEach Statements in SwiftUI

**Comment by wongzigii** [comment]
@BenShutt  PR is welcomed.

**Comment by BenShutt** [comment]
Hey @wongzigii, my _workaround_ what mapping the `JSON` to its string representation and using the `Hashable` of that. It didn't feel robust so I was/am curious to hear what you (all) suggest.

**Comment by Vyazovoy** [comment]
@BenShutt, your approach is pretty robust as long as you are using [sortedKeys](https://developer.apple.com/documentation/foundation/jsonserialization/writingoptions/2888322-sortedkeys) writing option for the data generation.

**Comment by guykogus** [comment]
If you'd like to swap over to a different implementation, I've built something very similar that is `Hashable`
https://github.com/guykogus/SwifterJSON
